### PR TITLE
inital scaffolding of nested subcommands for discussion

### PR DESCRIPTION
**Pull request by jbaxleyiii**
@queerviolet @JakeDawkins I'd love your thoughts on the setup to build out nested commands (i.e: `apollo schema push`)

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -76,6 +76,24 @@ impl Subcommand {
             Subcommand::Login(login) => login.run(),
             Subcommand::Print(print) => print.run(),
             Subcommand::Setup(setup) => setup.run(),
+            Subcommand::Schema { command } => match command {
```
Comment (queerviolet) [inline cli/src/cli.rs:79]: it seems kindof unfriendly to expect every subcommand implementer to do this?

File: cli/src/cli.rs [hunk 2]
```diff
@@ -62,6 +54,14 @@ pub enum Subcommand {
     #[structopt(name = "login")]
     ///  🔓  log in to apollo
     Login(commands::Login),
+
+    #[structopt(name = "schema")]
+    ///  ⚛   manage your schema
+    Schema {
+        #[structopt(subcommand)]
+        command: Option<commands::schema::SchemaCommand>,
```
Comment (queerviolet) [inline cli/src/cli.rs:62]: what happens if this isn't optional?

**Comment by queerviolet** [comment]
Something feels a little off about us having to manually invoke help. What happens if we don't make the command an option. Does that just give us the behavior we want (i.e. if it's not there, structopt prints help?)

**Comment by queerviolet** [comment]
Particularly as we look at adding nested subcommands, it seems like it would be nice not to have to replicate that check and re-invoke pattern at every subcommand and every level of the tree.

**Comment by jbaxleyiii** [comment]
@queerviolet yeah I agree with your comments which is where my general 😒 about this comes from. I'll try some other ideas but maybe we can pair on Friday on it?

**Comment by queerviolet** [comment]
Definitely down to pair!

File: cli/src/cli.rs [hunk 3]
```diff
@@ -76,6 +81,10 @@ impl Subcommand {
             Subcommand::Login(login) => login.run(),
             Subcommand::Print(print) => print.run(),
             Subcommand::Setup(setup) => setup.run(),
+            Subcommand::Schema(command) => match command {
```
Comment (queerviolet) [inline cli/src/cli.rs:84]: maybe move this into `schema/mod.rs`?

File: cli/src/commands/schema/push.rs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+use structopt::StructOpt;
+
+use crate::commands::Command;
+use crate::errors::{ExitCode, Fallible};
+
+#[derive(StructOpt)]
+pub struct Push {}
+
+impl Command for Push {
```
Comment (queerviolet) [inline cli/src/commands/schema/push.rs:9]: is the `Command` trait actually doing anything at this point? i introduced it so we wouldn't have to directly reference the implementation files from the command schema, but we're doing that now, so we may as well get rid of the trait

**Comment by jbaxleyiii** [comment]
I think this is a good enough direction for now. Going to close this but when we start building out subcommands its a good template to use

**Pull request closed without merging**